src/client/components/pages/admin-panel-search.tsx

```diff
@@ -0,0 +1,186 @@
+	 * sets it in the state to be passed down to AdminPanelSearchField and Pager components
+	 *
+	 * @param {string} query - Query string entered by user.
+	 */
+	handleSearch = (query: string) => {
+		this.setState({query});
+	};
+
+	/**
+	 * The Pager component deals with fetching the query from the server.
+	 * We use this callback to set the results on this component's state.
+	 *
+	 * @param {array} newResults - The array of results from the  query
+	 */
+	searchResultsCallback = (newResults: any[]) => {
+		this.setState({results: newResults});
+	};
+
+	/**
+	 * The Pager component is set up to react to browser history navigation (prev/next buttons),
+	 * and we use this callback to set the query and type on this component's state.
+	 *
+	 * @param {URLSearchParams} searchParams - The URL search parameters passed up from the pager component
+	 */
+	searchParamsChangeCallback = (searchParams: URLSearchParams) => {
+		let query;
+		if (searchParams.has('q')) {
+			query = searchParams.get('q');
+		}
+		if (query === this.state.query) {
+			return;
+		}
+		this.handleSearch(query);
+	};
+
+	/**
+	 * When we update the privileges of the user, we need to update the information in the results page
+	 * The updateResultsTrigger is passed as a prop in the Pager Component, whenever we want to
+	 * get updated results, we need to just use the triggerSearch function in the Pager Component. In order to
+	 * trigger that function without any change in query, we can just update this prop by flipping it between 0 and 1.
+	 * This will trigger the componentDidUpdate lifecycle method in Pager component and then we can run the
+	 * triggerSearch function
+	 */
+	updateResultsOnPrivsChange = () => {
+		const {updateResultsTrigger: trigger} = this.state;
+		this.setState({
+			updateResultsTrigger: 1 - trigger
+		});
+	};
+
+	/**
+	 * Renders the component: Search bar with results table located vertically
+	 * below it.
+	 *
+	 * @returns {object} - JSX to render.
+	 */
+	render() {
+		const {query, results, updateResultsTrigger} = this.state;
+		const querySearchParams = `q=${query}&type=editor`;
+		return (
+			<Card>
+				<Card.Header as="h2">
+					Admin Panel
+				</Card.Header>
+				<Card.Body>
+					<div id="pageWithPagination">
+						<AdminPanelSearchField
+							query={query}
+							onSearch={this.handleSearch}
+						/>
+						<AdminPanelSearchResults
+							results={this.state.results}
+							updateResultsOnPrivsChange={this.updateResultsOnPrivsChange}
+							user={this.props.user}
+						/>
+						<PagerElement
+							from={this.props.from}
+							nextEnabled={this.props.nextEnabled}
+							paginationUrl={this.paginationUrl}
+							querySearchParams={querySearchParams}
+							results={results}
+							searchParamsChangeCallback={this.searchParamsChangeCallback}
+							searchResultsCallback={this.searchResultsCallback}
+							size={this.props.resultsPerPage}
+							updateResultsTrigger={updateResultsTrigger}
```

Comment:
I have another mechanism to propose for this, that looks less like a workaround.
Instead of using this `updateResultsTrigger` to trigger a method in `componentDidUpdate`, you can save a reference to the Pager component in this AdminPanelSearchPage class (using the ref prop like so `ref={this.pagerRef}`), and then in `updateResultsOnPrivsChange` we can use that ref to call a public method of the Pager class directly: `this.pagerRef?.current?.triggerSearch()` <- this method in Pager default to using the current state values from `from` and `size`.

Note the syntax above uses `?.` too make sure there is a stored reference before calling the method, otherwise you can get some nasty "method XXX does not exist on undefined" errors.

You'll need instantiate that ref variable in the AdminPanelSearchPage constructor (`this.pagerRef = React.createRef();`)

Reply:
Thanks for this suggestion. I have made a [PR](https://github.com/metabrainz/bookbrainz-site/pull/998) with these changes.